resource-managers/yarn/src/test/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackendSuite.scala

```diff
@@ -32,15 +33,33 @@ import org.apache.spark.ui.TestFilter
 
 class YarnSchedulerBackendSuite extends SparkFunSuite with MockitoSugar with LocalSparkContext {
 
+  private var yarnSchedulerBackend: YarnSchedulerBackend = _
+
+  override def afterEach() {
+    try {
+      yarnSchedulerBackend.stop()
+    } finally {
+      super.afterEach()
+    }
+  }
+
   test("RequestExecutors reflects node blacklist and is serializable") {
     sc = new SparkContext("local", "YarnSchedulerBackendSuite")
-    val sched = mock[TaskSchedulerImpl]
-    when(sched.sc).thenReturn(sc)
-    val yarnSchedulerBackend = new YarnSchedulerBackend(sched, sc) {
+    // Subclassing the TaskSchedulerImpl here instead of using Mockito. For details see SPARK-26891.
+    val sched = new TaskSchedulerImpl(sc) {
+      val blacklistedNodes = new AtomicReference[Set[String]]()
+
+      def setNodeBlacklist(nodeBlacklist: Set[String]): Unit = blacklistedNodes.set(nodeBlacklist)
+
+      override def nodeBlacklist(): Set[String] = blacklistedNodes.get()
+    }
+
+    val yarnSchedulerBackendExtended = new YarnSchedulerBackend(sched, sc) {
```

Comment:
Do you need the extra variable here, vs assigning to `yarnSchedulerBackend`? I don't see that they are used separately.

Reply:
It is needed because of the different type: the `yarnSchedulerBackend` type is `YarnSchedulerBackend` but `yarnSchedulerBackendExtended` type is an anonim subclass of `YarnSchedulerBackend` with the extra def `setNodeBlacklist`. On `yarnSchedulerBackend` I cannot call this extra method.

Reply:
If so then how is it assigned in the next line? a subclass of `YarnSchedulerBackend` is still assignable to `YarnSchedulerBackend`. I might be missing something obvious here.

Reply:
It is assignable as `yarnSchedulerBackendExtended` is an instance of `YarnSchedulerBackend` too, although not a direct one.